services/worker/src/worker/job_runners/dataset/dataset_job_runner.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+from pathlib import Path
+
+from libcommon.processing_graph import ProcessingStep
+from libcommon.utils import JobInfo
+
+from worker.common_exceptions import ParameterMissingError
+from worker.config import AppConfig
+from worker.job_runner import JobRunner
+from worker.job_runners._datasets_based_job_runner import DatasetsBasedJobRunner
+
+
+class DatasetJobRunner(JobRunner):
+    dataset: str
+
+    def __init__(
+        self,
+        job_info: JobInfo,
+        app_config: AppConfig,
+        processing_step: ProcessingStep,
+    ) -> None:
+        super().__init__(job_info=job_info, app_config=app_config, processing_step=processing_step)
+        if job_info["params"]["dataset"] is None:
+            raise ParameterMissingError("'dataset' parameter is required")
+        self.dataset = job_info["params"]["dataset"]
+
+
+class DatasetCachedJobRunner(DatasetsBasedJobRunner, DatasetJobRunner):
+    def __init__(
+        self,
+        job_info: JobInfo,
+        app_config: AppConfig,
+        processing_step: ProcessingStep,
+        hf_datasets_cache: Path,
+    ) -> None:
+        DatasetsBasedJobRunner.__init__(
+            self=self,
+            job_info=job_info,
+            app_config=app_config,
+            processing_step=processing_step,
+            hf_datasets_cache=hf_datasets_cache,
+        )
+        DatasetJobRunner.__init__(
```

Comment:
What is the difference between calling super().__init__() and, like here, to call explicitly the parent class names? Is one of them super() too? 
I'm not familiar with double inheritance.

Reply:
For this specific case the problem was having different parameters/types in the parent constructors, and calling to super().__init__() would expect to have the same parameters sign. One solution was to have *args and **kwargs in ancestors but I didn't want to complicate them. But maybe there is a better way to handle it.

Reply:
OK, no idea. Thanks